### DISCUSSION TITLE
Fatal: Reference repository is shallow

### DISCUSSION THREAD

**Discussion by jamieburchell** [deployphp/deployer#2741] [Help needed]
Hello

I've been running Deployer 6.x with Git 1.8.3.1 and today I copied over my deployments in their entirety to a new server which has Git 2.2.7.0.

When I ran a dep deploy I received a "Fatal: Reference repository is shallow" error message, before the clone was completed successfully.

I checked the previous deploys and these had a .git/shallow file present, but this latest deploy doesn't.

I don't understand what this is about because the Deployer version is the same and by default the update code is using the same depth as before?

Jamie

**Comment by antonmedv** [comment]
Maybe copy process was not done right?

**Reply by jamieburchell** [reply]
It's an exact copy. I've confirmed the .git/shadow exists on the "old" server and they were copied over to the new server. The latest dep deploy on the new server doesn't include the shadow file, but I believe that's a change in the new Git version.

The copy was performed with rsync -avzXA as sudo on both sides.

**Reply by jamieburchell** [reply]
If you think about it, it's not really about the copy. The old deploys were shallow copies as indicated by the .git/shallow file. The new deploy is telling me that the reference repo was shallow (correct). The new deploy on the new server is not shallow.

**Reply by antonmedv** [reply]
You can debug and see exactly what git command is executed via -vvv option. Try to see what is executed.

**Reply by jamieburchell** [reply]
It's running this on the new server:

`/usr/bin/git clone -b "testing" --recursive  --reference /srv/example.test/releases/19 --dissociate git@bitbucket.org:test/example.test.git  /srv/example.test/releases/20 2>&1`

Same deploy file on old server:

`/usr/bin/git clone -b "testing" --depth 1 --recursive  git@bitbucket.org:test/example.test.git /srv/example.test/releases/20 2>&1`

So I guess Deployer behaves differently depending on Git version (looks like git_cache) and so given old deploys were with one approach and new deploys with another, there's a transition "issue".

There could probably be a check added to see if the reference repo is a shallow copy or not before trying to use it as it can't be assumed just from the version of Git installed.

**Reply by antonmedv** [reply]
In v7 Deployer uses a different approach. Git  archive
